pkg/network/usm/ebpf_javatls.go

```diff
@@ -36,195 +36,176 @@ const (
-	// if matching the agent-usm.jar would or not injected
-	javaAgentAllowRegex *regexp.Regexp
-	javaAgentBlockRegex *regexp.Regexp
 )
 
-type JavaTLSProgram struct {
+type javaTLSProgram struct {
 	cfg            *config.Config
-	manager        *nettelemetry.Manager
 	processMonitor *monitor.ProcessMonitor
 	cleanupExec    func()
+
+	// tracerJarPath path to the USM agent TLS tracer.
+	tracerJarPath string
+
+	// tracerArguments default arguments passed to the injected agent-usm.jar
+	tracerArguments string
+
+	// injectionAllowRegex is matched against /proc/pid/cmdline, to determine if we should attach to the process.
+	injectionAllowRegex *regexp.Regexp
+	// injectionAllowRegex is matched against /proc/pid/cmdline, to determine if we should deny attachment to the process.
+	injectionBlockRegex *regexp.Regexp
+
+	procRoot string
 }
 
 // Static evaluation to make sure we are not breaking the interface.
-var _ subprogram = &JavaTLSProgram{}
+var _ subprogram = &javaTLSProgram{}
 
-func GetJavaTlsTailCallRoutes() []manager.TailCallRoute {
+func getJavaTlsTailCallRoutes() []manager.TailCallRoute {
 	return []manager.TailCallRoute{
 		{
 			ProgArrayName: eRPCHandlersMap,
-			Key:           SyncPayload,
+			Key:           syncPayload,
 			ProbeIdentificationPair: manager.ProbeIdentificationPair{
-				EBPFFuncName: "kprobe_handle_sync_payload",
+				EBPFFuncName: handleSyncPayloadKprobeName,
 			},
 		},
 		{
 			ProgArrayName: eRPCHandlersMap,
-			Key:           CloseConnection,
+			Key:           closeConnection,
 			ProbeIdentificationPair: manager.ProbeIdentificationPair{
-				EBPFFuncName: "kprobe_handle_close_connection",
+				EBPFFuncName: handleCloseConnectionKprobeName,
 			},
 		},
 		{
 			ProgArrayName: eRPCHandlersMap,
-			Key:           ConnectionByPeer,
+			Key:           connectionByPeer,
 			ProbeIdentificationPair: manager.ProbeIdentificationPair{
-				EBPFFuncName: "kprobe_handle_connection_by_peer",
+				EBPFFuncName: handleConnectionByPeerKprobeName,
 			},
 		},
 		{
 			ProgArrayName: eRPCHandlersMap,
-			Key:           AsyncPayload,
+			Key:           asyncPayload,
 			ProbeIdentificationPair: manager.ProbeIdentificationPair{
-				EBPFFuncName: "kprobe_handle_async_payload",
+				EBPFFuncName: handleAsyncPayloadKprobeName,
 			},
 		},
 	}
 }
 
-func IsJavaSubprogramEnabled(c *config.Config) bool {
+func newJavaTLSProgram(c *config.Config) *javaTLSProgram {
+	var res *javaTLSProgram
+	defer func() {
+		// If we didn't set res, then java tls initialization failed.
+		if res == nil {
+			log.Info("java tls is not enabled")
+		}
+	}()
+
 	if !c.EnableJavaTLSSupport || !c.EnableHTTPSMonitoring || !http.HTTPSSupported(c) {
-		return false
+		return nil
 	}
 
-	javaUSMAgentJarPath = filepath.Join(c.JavaDir, agentUSMJar)
-	jar, err := os.Open(javaUSMAgentJarPath)
-	if err != nil {
+	javaUSMAgentJarPath := filepath.Join(c.JavaDir, agentUSMJar)
+	if _, err := os.Stat(javaUSMAgentJarPath); err != nil {
```

Comment:
https://github.com/DataDog/datadog-agent/pull/14620/files#r1065824758

Reply:
I wish that would have become a comment in the code from the first place

I'll revert and add comment